userspace/libsinsp/event.h

```diff
@@ -343,6 +344,57 @@ class SINSP_PUBLIC sinsp_evt : public gen_event
 	*/
 	sinsp_evt_param* get_param(uint32_t id);
 
+	/*!
+	  \brief Get the value of a fixed param (fixed length type or packed struct).
+
+	  \param id The parameter number.
+	*/
+	template<class T>
+	inline T get_param(uint32_t id)
+	{
+		T ret;
+		const sinsp_evt_param *param = get_param(id);
+
+		if (param->m_len != sizeof(T))
+		{
+			// By moving this error string building operation to a separate function
+			// the compiler is more likely to inline this entire function.
+			// This is important since get_param<> is called in the hot path.
+			throw sinsp_exception(invalid_param_len_error(id, sizeof(T)));
+		}
+
+		memcpy(&ret, param->m_val, sizeof(T));
+
+		return ret;
+	}
+
+	/*!
+	  \brief Get the value of a string param as a const char*.
+
+	  \param id The parameter number.
+	*/
+
+	inline const char* get_param_const_char(uint32_t id)
```

Comment:
We have experimented a bit with @gnosek and have a path forward to remove this and have only one template and also solve more UB. I promised I'll remove this :)